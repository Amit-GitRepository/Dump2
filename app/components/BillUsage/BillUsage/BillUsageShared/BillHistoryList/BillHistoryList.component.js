import moment from '../../../../../utils/moment.util';
import noop from 'lodash/noop';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './BillHistoryList.style';
import {AmountText, Button, Checkbox, ISText} from '../../../../Shared';
import {translate} from '../../../../../language/i18n/helper';
import {View} from 'react-native';

class BillHistoryList extends Component {

  billDetailClick = (invoiceNo) => () => this.props.onBillDetailsClick(invoiceNo);

  render () {
    const {dueBills, style, onPaymentHistoryClick, onBillToggle, isCancelled} = this.props;
    const currentDate = moment();
    return (
      <View style={[styles.container, style]}>
        {isCancelled && <View style={styles.titleContainer}><ISText type='BOLD' style={styles.title}>{translate('BILLS_USAGE_BILL_DETAILS')}</ISText>
        </View>}
        {dueBills && dueBills.length > 0 ?
          dueBills.map((bill, index) => {
            const billDate = new moment(bill.invoiceCycle, 'DD/MM/YYYY');
            const billDueDate = new moment(bill.dueDate);
            const overDueDays = currentDate.diff(billDueDate, 'days');
            return (
              <View style={styles.cardContainer} key={`billingCard_${index}`}>
                <View style={styles.leftSection}>
                  <ISText type='BOLD' style={styles.billMonthText}>{billDate.format('MMMM')}</ISText>
                  <ISText type='BOLD' style={styles.dueDateText}>{(overDueDays > 0) ? `${translate('BILLS_USAGE_OVERDUE', {count: overDueDays, days: overDueDays})}` : `${translate('BILLS_USAGE_DUE_DATE')}: ${billDueDate.format('DD MMM YY')}`}</ISText>
                </View>
                <View style={styles.rightSection}>
                  <View style={styles.amountSection}>
                    <AmountText value={bill.unpaidAmount} isTotalText={false}/>
                    <Checkbox style={styles.checkbox} onChange={onBillToggle(bill)} isChecked = {bill.isChecked}/>
                  </View>
                  <Button type='inline' text={translate('BILLS_USAGE_DETAILS')} style={styles.billDetailButton} textStyle={styles.billDetailText} onPress = {this.billDetailClick({invoiceNo: bill.invoiceNo})}/>
                </View>
              </View>
            );
          }) : <ISText type='BOLD' style = {styles.noBillPlaceHolderText}>{translate('BILLS_USAGE_BILL_DETAILS_PLACEHOLDER')}</ISText>
        }
        <Button type='inline' text={translate('BILLS_USAGE_PAYMENT_HISTORY')} textStyle={styles.otherButton} onPress={onPaymentHistoryClick}/>
      </View>
    );
  }
}

BillHistoryList.defaultProps = {
  dueBills: [],
  onBillDetailsClick: noop,
  onPaymentHistoryClick: noop,
  onBillToggle: noop,
  isCancelled: false,
  style: {}
};

BillHistoryList.propTypes = {
  dueBills: Proptypes.array,
  onBillDetailsClick: Proptypes.func,
  onPaymentHistoryClick: Proptypes.func,
  style: Proptypes.object,
  onBillToggle: Proptypes.func,
  isCancelled: Proptypes.bool
};

export default BillHistoryList;
