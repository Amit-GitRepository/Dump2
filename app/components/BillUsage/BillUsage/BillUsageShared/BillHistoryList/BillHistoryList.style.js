import {colors, fonts} from '../../../../../themes/constants.styles';

const sectionStyles = {
  flex: 1,
  justifyContent: 'flex-start'
};

export default {
  container: {
    flexGrow: 1
  },
  cardContainer: {
    flexGrow: 1,
    flexDirection: 'row',
    marginHorizontal: 10,
    paddingVertical: 10,
    borderBottomWidth: 1.5,
    borderColor: colors.PRIMARY_BG_SEPARATOR
  },
  titleContainer: {
    marginHorizontal: 10,
    borderBottomWidth: 1.5,
    borderColor: colors.PRIMARY_BG_SEPARATOR,
    alignItems: 'center',
    paddingVertical: 10
  },
  leftSection: {
    ...sectionStyles
  },
  rightSection: {
    ...sectionStyles,
    alignItems: 'flex-end'
  },
  billMonthText: {
    color: colors.PRIMARY_SUBMENU,
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  dueDateText: {
    fontSize: fonts.FONT_SIZE_SMALL
  },
  billDetailButton: {
    alignSelf: 'flex-end',
    paddingVertical: 0
  },
  billDetailText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    padding: 0
  },
  amountSection: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  amountText: {
    paddingRight: 5,
    fontSize: fonts.FONT_SIZE_LARGE
  },
  paymentHistoryButton: {},
  paymentHistoryText: {},
  checkbox: {
    marginLeft: 5
  },
  noBillPlaceHolderText: {
    color: colors.PRIMARY_DISABLED_BG_TEXT,
    alignSelf: 'center'
  },
  title: {
    fontSize: fonts.FONT_SIZE_NORMAL,
    color: colors.SECONDARY_BG_TEXT_CONTRAST
  }
};
