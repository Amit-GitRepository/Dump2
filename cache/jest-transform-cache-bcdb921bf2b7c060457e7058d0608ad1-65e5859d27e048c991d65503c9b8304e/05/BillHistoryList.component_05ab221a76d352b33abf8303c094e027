Object.defineProperty(exports,"__esModule",{value:true});var cov_6dqu3epko=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/BillHistoryList/BillHistoryList.component.js',hash='ab82ab10fe00b15fdc077eec910a1243170ba9ee',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/BillHistoryList/BillHistoryList.component.js',statementMap:{'0':{start:{line:12,column:35},end:{line:12,column:81}},'1':{start:{line:12,column:41},end:{line:12,column:81}},'2':{start:{line:15,column:80},end:{line:15,column:90}},'3':{start:{line:16,column:24},end:{line:16,column:32}},'4':{start:{line:17,column:4},end:{line:45,column:6}},'5':{start:{line:23,column:29},end:{line:23,column:72}},'6':{start:{line:24,column:32},end:{line:24,column:56}},'7':{start:{line:25,column:32},end:{line:25,column:69}},'8':{start:{line:26,column:12},end:{line:40,column:14}},'9':{start:{line:49,column:0},end:{line:56,column:2}},'10':{start:{line:58,column:0},end:{line:65,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:20},end:{line:12,column:21}},loc:{start:{line:12,column:35},end:{line:12,column:81}},line:12},'1':{name:'(anonymous_1)',decl:{start:{line:12,column:35},end:{line:12,column:36}},loc:{start:{line:12,column:41},end:{line:12,column:81}},line:12},'2':{name:'(anonymous_2)',decl:{start:{line:14,column:2},end:{line:14,column:3}},loc:{start:{line:14,column:12},end:{line:46,column:3}},line:14},'3':{name:'(anonymous_3)',decl:{start:{line:22,column:23},end:{line:22,column:24}},loc:{start:{line:22,column:40},end:{line:41,column:11}},line:22}},branchMap:{'0':{loc:{start:{line:19,column:9},end:{line:20,column:15}},type:'binary-expr',locations:[{start:{line:19,column:9},end:{line:19,column:20}},{start:{line:19,column:24},end:{line:20,column:15}}],line:19},'1':{loc:{start:{line:21,column:9},end:{line:41,column:134}},type:'cond-expr',locations:[{start:{line:22,column:10},end:{line:41,column:12}},{start:{line:41,column:15},end:{line:41,column:134}}],line:21},'2':{loc:{start:{line:21,column:9},end:{line:21,column:40}},type:'binary-expr',locations:[{start:{line:21,column:9},end:{line:21,column:17}},{start:{line:21,column:21},end:{line:21,column:40}}],line:21},'3':{loc:{start:{line:30,column:66},end:{line:30,column:241}},type:'cond-expr',locations:[{start:{line:30,column:86},end:{line:30,column:164}},{start:{line:30,column:167},end:{line:30,column:241}}],line:30}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0},f:{'0':0,'1':0,'2':0,'3':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _moment=require('../../../../../utils/moment.util');var _moment2=_interopRequireDefault(_moment);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _BillHistoryList=require('./BillHistoryList.style');var _BillHistoryList2=_interopRequireDefault(_BillHistoryList);
var _Shared=require('../../../../Shared');
var _helper=require('../../../../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

BillHistoryList=function(_Component){_inherits(BillHistoryList,_Component);function BillHistoryList(){var _ref;var _temp,_this,_ret;_classCallCheck(this,BillHistoryList);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=BillHistoryList.__proto__||Object.getPrototypeOf(BillHistoryList)).call.apply(_ref,[this].concat(args))),_this),_this.

billDetailClick=function(invoiceNo){cov_6dqu3epko.f[0]++;cov_6dqu3epko.s[0]++;return function(){cov_6dqu3epko.f[1]++;cov_6dqu3epko.s[1]++;return _this.props.onBillDetailsClick(invoiceNo);};},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(BillHistoryList,[{key:'render',value:function render()

{var _this2=this;cov_6dqu3epko.f[2]++;var _ref2=(cov_6dqu3epko.s[2]++,
this.props),dueBills=_ref2.dueBills,style=_ref2.style,onPaymentHistoryClick=_ref2.onPaymentHistoryClick,onBillToggle=_ref2.onBillToggle,isCancelled=_ref2.isCancelled;
var currentDate=(cov_6dqu3epko.s[3]++,(0,_moment2.default)());cov_6dqu3epko.s[4]++;
return(
_react2.default.createElement(_reactNative.View,{style:[_BillHistoryList2.default.container,style]},
(cov_6dqu3epko.b[0][0]++,isCancelled)&&(cov_6dqu3epko.b[0][1]++,_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.titleContainer},_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_BillHistoryList2.default.title},(0,_helper.translate)('BILLS_USAGE_BILL_DETAILS')))),

(cov_6dqu3epko.b[2][0]++,dueBills)&&(cov_6dqu3epko.b[2][1]++,dueBills.length>0)?(cov_6dqu3epko.b[1][0]++,
dueBills.map(function(bill,index){cov_6dqu3epko.f[3]++;
var billDate=(cov_6dqu3epko.s[5]++,new _moment2.default(bill.invoiceCycle,'DD/MM/YYYY'));
var billDueDate=(cov_6dqu3epko.s[6]++,new _moment2.default(bill.dueDate));
var overDueDays=(cov_6dqu3epko.s[7]++,currentDate.diff(billDueDate,'days'));cov_6dqu3epko.s[8]++;
return(
_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.cardContainer,key:'billingCard_'+index},
_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.leftSection},
_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_BillHistoryList2.default.billMonthText},billDate.format('MMMM')),
_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_BillHistoryList2.default.dueDateText},overDueDays>0?(cov_6dqu3epko.b[3][0]++,''+(0,_helper.translate)('BILLS_USAGE_OVERDUE',{count:overDueDays,days:overDueDays})):(cov_6dqu3epko.b[3][1]++,(0,_helper.translate)('BILLS_USAGE_DUE_DATE')+': '+billDueDate.format('DD MMM YY')))),

_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.rightSection},
_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.amountSection},
_react2.default.createElement(_Shared.AmountText,{value:bill.unpaidAmount,isTotalText:false}),
_react2.default.createElement(_Shared.Checkbox,{style:_BillHistoryList2.default.checkbox,onChange:onBillToggle(bill),isChecked:bill.isChecked})),

_react2.default.createElement(_Shared.Button,{type:'inline',text:(0,_helper.translate)('BILLS_USAGE_DETAILS'),style:_BillHistoryList2.default.billDetailButton,textStyle:_BillHistoryList2.default.billDetailText,onPress:_this2.billDetailClick({invoiceNo:bill.invoiceNo})}))));



})):(cov_6dqu3epko.b[1][1]++,_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_BillHistoryList2.default.noBillPlaceHolderText},(0,_helper.translate)('BILLS_USAGE_BILL_DETAILS_PLACEHOLDER'))),

_react2.default.createElement(_Shared.Button,{type:'inline',text:(0,_helper.translate)('BILLS_USAGE_PAYMENT_HISTORY'),textStyle:_BillHistoryList2.default.otherButton,onPress:onPaymentHistoryClick})));


}}]);return BillHistoryList;}(_react.Component);cov_6dqu3epko.s[9]++;


BillHistoryList.defaultProps={
dueBills:[],
onBillDetailsClick:_noop2.default,
onPaymentHistoryClick:_noop2.default,
onBillToggle:_noop2.default,
isCancelled:false,
style:{}};cov_6dqu3epko.s[10]++;


BillHistoryList.propTypes={
dueBills:_propTypes2.default.array,
onBillDetailsClick:_propTypes2.default.func,
onPaymentHistoryClick:_propTypes2.default.func,
style:_propTypes2.default.object,
onBillToggle:_propTypes2.default.func,
isCancelled:_propTypes2.default.bool};exports.default=


BillHistoryList;