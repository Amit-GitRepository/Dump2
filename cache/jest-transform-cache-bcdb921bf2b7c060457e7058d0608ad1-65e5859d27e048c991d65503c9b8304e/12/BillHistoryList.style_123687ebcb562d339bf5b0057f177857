Object.defineProperty(exports,"__esModule",{value:true});var cov_2h4283usov=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/BillHistoryList/BillHistoryList.style.js',hash='e1d09a4872dd84384af3845fd469a618db0c2d13',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/BillHistoryList/BillHistoryList.style.js',statementMap:{'0':{start:{line:3,column:22},end:{line:6,column:1}}},fnMap:{},branchMap:{},s:{'0':0},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _constants=require('../../../../../themes/constants.styles');

var sectionStyles=(cov_2h4283usov.s[0]++,{
flex:1,
justifyContent:'flex-start'});exports.default=


{
container:{
flexGrow:1},

cardContainer:{
flexGrow:1,
flexDirection:'row',
marginHorizontal:10,
paddingVertical:10,
borderBottomWidth:1.5,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR},

titleContainer:{
marginHorizontal:10,
borderBottomWidth:1.5,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR,
alignItems:'center',
paddingVertical:10},

leftSection:_extends({},
sectionStyles),

rightSection:_extends({},
sectionStyles,{
alignItems:'flex-end'}),

billMonthText:{
color:_constants.colors.PRIMARY_SUBMENU,
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

dueDateText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL},

billDetailButton:{
alignSelf:'flex-end',
paddingVertical:0},

billDetailText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL,
padding:0},

amountSection:{
flexDirection:'row',
alignItems:'center',
justifyContent:'center'},

amountText:{
paddingRight:5,
fontSize:_constants.fonts.FONT_SIZE_LARGE},

paymentHistoryButton:{},
paymentHistoryText:{},
checkbox:{
marginLeft:5},

noBillPlaceHolderText:{
color:_constants.colors.PRIMARY_DISABLED_BG_TEXT,
alignSelf:'center'},

title:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST}};