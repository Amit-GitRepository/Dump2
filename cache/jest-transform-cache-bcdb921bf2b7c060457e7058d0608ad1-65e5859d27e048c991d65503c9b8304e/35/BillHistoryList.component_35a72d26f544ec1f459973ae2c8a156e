Object.defineProperty(exports,"__esModule",{value:true});var cov_21usae7ddl=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/BillHistoryList.component.js',hash='69268bf7fe8646bfbf5ca6b4b5c805b830100b82',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/BillHistoryList.component.js',statementMap:{'0':{start:{line:13,column:24},end:{line:42,column:1}},'1':{start:{line:14,column:22},end:{line:14,column:30}},'2':{start:{line:15,column:2},end:{line:41,column:4}},'3':{start:{line:19,column:27},end:{line:19,column:70}},'4':{start:{line:20,column:30},end:{line:20,column:54}},'5':{start:{line:21,column:30},end:{line:21,column:67}},'6':{start:{line:22,column:10},end:{line:36,column:12}},'7':{start:{line:44,column:0},end:{line:50,column:2}},'8':{start:{line:52,column:0},end:{line:58,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:13,column:24},end:{line:13,column:25}},loc:{start:{line:13,column:104},end:{line:42,column:1}},line:13},'1':{name:'(anonymous_1)',decl:{start:{line:18,column:21},end:{line:18,column:22}},loc:{start:{line:18,column:38},end:{line:37,column:9}},line:18}},branchMap:{'0':{loc:{start:{line:26,column:64},end:{line:26,column:239}},type:'cond-expr',locations:[{start:{line:26,column:84},end:{line:26,column:162}},{start:{line:26,column:165},end:{line:26,column:239}}],line:26}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0},f:{'0':0,'1':0},b:{'0':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _AmountText=require('../../../Shared/AmountText/AmountText.component');var _AmountText2=_interopRequireDefault(_AmountText);
var _Button=require('../../../Shared/Button/Button.component');var _Button2=_interopRequireDefault(_Button);
var _Checkbox=require('../../../Shared/Checkbox/Checkbox.component');var _Checkbox2=_interopRequireDefault(_Checkbox);
var _ISText=require('../../../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _moment=require('../../../../utils/moment.util');var _moment2=_interopRequireDefault(_moment);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _BillHistoryList=require('./BillHistoryList.style');var _BillHistoryList2=_interopRequireDefault(_BillHistoryList);
var _helper=require('../../../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}cov_21usae7ddl.s[0]++;

var BillHistoryList=function BillHistoryList(_ref){var dueBills=_ref.dueBills,style=_ref.style,onPaymentHistoryClick=_ref.onPaymentHistoryClick,onBillDetailsClick=_ref.onBillDetailsClick,onBillToggle=_ref.onBillToggle;cov_21usae7ddl.f[0]++;
var currentDate=(cov_21usae7ddl.s[1]++,(0,_moment2.default)());cov_21usae7ddl.s[2]++;
return(
_react2.default.createElement(_reactNative.View,{style:[_BillHistoryList2.default.container,style]},

dueBills.map(function(bill,index){cov_21usae7ddl.f[1]++;
var billDate=(cov_21usae7ddl.s[3]++,new _moment2.default(bill.invoiceCycle,'DD/MM/YYYY'));
var billDueDate=(cov_21usae7ddl.s[4]++,new _moment2.default(bill.dueDate));
var overDueDays=(cov_21usae7ddl.s[5]++,currentDate.diff(billDueDate,'days'));cov_21usae7ddl.s[6]++;
return(
_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.cardContainer,key:'billingCard_'+index},
_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.leftSection},
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:_BillHistoryList2.default.billMonthText},billDate.format('MMMM')),
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:_BillHistoryList2.default.dueDateText},overDueDays>0?(cov_21usae7ddl.b[0][0]++,''+(0,_helper.translate)('BILLS_USAGE_OVERDUE',{count:overDueDays,days:overDueDays})):(cov_21usae7ddl.b[0][1]++,(0,_helper.translate)('BILLS_USAGE_DUE_DATE')+': '+billDueDate.format('DD MMM YY')))),

_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.rightSection},
_react2.default.createElement(_reactNative.View,{style:_BillHistoryList2.default.amountSection},
_react2.default.createElement(_AmountText2.default,{value:bill.unpaidAmount,isTotalText:false}),
_react2.default.createElement(_Checkbox2.default,{style:_BillHistoryList2.default.checkbox,onChange:onBillToggle(bill.invoiceNo)})),

_react2.default.createElement(_Button2.default,{type:'inline',text:(0,_helper.translate)('BILLS_USAGE_DETAILS'),style:_BillHistoryList2.default.billDetailButton,textStyle:_BillHistoryList2.default.billDetailText,onPress:onBillDetailsClick(bill.invoiceNo)}))));



}),

_react2.default.createElement(_Button2.default,{type:'inline',text:(0,_helper.translate)('BILLS_USAGE_PAYMENT_HISTORY'),textStyle:_BillHistoryList2.default.otherButton,onPress:onPaymentHistoryClick})));


};cov_21usae7ddl.s[7]++;

BillHistoryList.defaultProps={
dueBills:[],
onBillDetailsClick:_noop2.default,
onPaymentHistoryClick:_noop2.default,
onBillToggle:_noop2.default,
style:{}};cov_21usae7ddl.s[8]++;


BillHistoryList.propTypes={
dueBills:_propTypes2.default.array,
onBillDetailsClick:_propTypes2.default.func,
onPaymentHistoryClick:_propTypes2.default.func,
onBillToggle:_propTypes2.default.func,
style:_propTypes2.default.object};exports.default=


BillHistoryList;