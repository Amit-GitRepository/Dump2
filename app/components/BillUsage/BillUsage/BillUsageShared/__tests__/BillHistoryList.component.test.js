import BillHistoryList from '../BillHistoryList/BillHistoryList.component';
import React from 'react'; // Enzyme adapter for React 16
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('BillHistoryList component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<BillHistoryList />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
