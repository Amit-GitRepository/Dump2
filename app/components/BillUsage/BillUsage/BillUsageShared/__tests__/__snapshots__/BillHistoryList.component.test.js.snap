// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillHistoryList component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexGrow": 1,
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "color": "#B9C1C9",
        "fontFamily": "SukhumvitSet-Bold",
      }
    }
  >
    No overdue bills
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderRadius": 50,
        "borderWidth": 0,
        "opacity": 1,
        "padding": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
              },
              Array [
                Object {
                  "color": "#F93E60",
                  "fontSize": 16,
                },
                Object {},
              ],
              Object {
                "fontFamily": "SukhumvitSet-Bold",
              },
            ]
          }
        >
          View payment history
        </Text>
      </View>
    </View>
  </View>
</View>
`;
